### ISSUE TITLE
`const_item_mutation` has false positives with `&mut &T`

### ISSUE THREAD

**Issue by jyn514** [rust-lang/rust#93388]
I tried this code:

```rust
    pub const SERVER_CERT: &[u8] = include_bytes!("./ca/server.pem");
    let certs = rustls_pemfile::certs(&mut SERVER_CERT).unwrap();
```

I expected to see this happen: No warning, because it's not actually possible for `SERVER_CERT` to be mutated, even if it weren't a constant.

Instead, this happened:
```
warning: taking a mutable reference to a `const` item
  --> src/lib.rs:42:39
   |
42 |     let certs = rustls_pemfile::certs(&mut SERVER_CERT).unwrap();
   |                                       ^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(const_item_mutation)]` on by default
   = note: each usage of a `const` item creates a new temporary
   = note: the mutable reference will refer to this temporary, not the original `const` item
note: `const` item defined here
  --> src/lib.rs:11:1
   |
11 | pub const SERVER_CERT: &[u8] = include_bytes!("../ca/server.pem");
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### Meta

`rustc --version --verbose`:
```
rustc 1.60.0-nightly (1bd4fdc94 2022-01-12)
binary: rustc
commit-hash: 1bd4fdc943513e1004f498bbf289279c9784fc6f
commit-date: 2022-01-12
host: x86_64-unknown-linux-gnu
release: 1.60.0-nightly
LLVM version: 13.0.0
```

**Comment by jyn514** [comment]
The signature for `certs` is `pub fn certs(rd: &mut dyn BufRead) -> Result<Vec<Vec<u8>>, Error>`: https://docs.rs/rustls-pemfile/0.2.1/rustls_pemfile/fn.certs.html

**Comment by nico-abram** [comment]
Does copying the &[u8] to a local not work?

```rs
fn main() {
    pub const SERVER_CERT: &[u8] = &[1, 2, 3];
    fn f(_: &mut dyn std::io::BufRead) {}
    
    // warning: taking a mutable reference to a `const` item
    f(&mut SERVER_CERT);
    
    // No warning
    let mut cert = SERVER_CERT;
    f(&mut cert);
}
```

**Comment by jyn514** [comment]
ah sure, there are workarounds - but I still think the lint is a false positive.

**Comment by Aaron1011** [comment]
> No warning, because it's not actually possible for SERVER_CERT to be mutated, even if it weren't a constant.

This is not true - you could attempt to write a different `&[u8]` to the mutable reference, which would actually just write to the temporary. The lint is working as intended here - `&mut SERVER_CERT`  makes it look like you have the ability to modify the global `SERVER_CERT`, which is not actually the case.
